src/main/groovy/nebula/plugin/dependencylock/tasks/GenerateLockTask.groovy

```diff
@@ -92,9 +95,9 @@ class GenerateLockTask extends AbstractLockTask {
             if (Configuration.class.declaredMethods.any { it.name == 'isCanBeResolved' }) {
                 project.configurations.findAll {
                     if (taskProject == project) {
-                        it.canBeResolved
+                        it.canBeResolved && !ConfigurationFilters.safelyHasAResolutionAlternative(it)
                     } else {
-                        it.canBeResolved && it.canBeConsumed
+                        it.canBeResolved && it.canBeConsumed && !ConfigurationFilters.safelyHasAResolutionAlternative(it)
```

Comment:
Here's one of the most interesting sections. In this method `lockableConfigurations`, we now also use `ConfigurationFilters` to check if the new method involving resolution alternatives exists & has content. If there are alternatives to this configuration for resolution, then we do not mark it as lockable